### PR TITLE
Added share post feature

### PR DESCRIPTION
**Pull request by adrean-hipe**
### OVERVIEW

- added share method in PostController
- added service for sharing posts
- added migration to insert new column in the user_posts table name original_post_id
- added connection to the UserPost model
- modified blades to display if the post is shared
- added share button to the post actions
- added route for share method (fixed merge conflict)

### NOTES

- php artisan migrate

### SNIPPETS

shared post
image

original post
image

### CODE REVIEW

**Review by bpocrafael — APPROVED** [review]

File: laravel-app/app/Interfaces/PostServiceInterface.php [hunk 1]
```diff
@@ -8,8 +8,14 @@
 interface PostServiceInterface
 {
     /**
-     * @param User $user
      * @param array<mixed, mixed> $validatedData
```
Comment (hipe01mark) [inline laravel-app/app/Interfaces/PostServiceInterface.php:11]: removee!!!!

File: laravel-app/app/Interfaces/PostServiceInterface.php [hunk 2]
```diff
@@ -8,8 +8,14 @@
...
      */
     public function createPost(User $user, array $validatedData): UserPost;
+
+    /**
+     * @param array<string, mixed> $validatedData
```
Comment (hipe01mark) [inline laravel-app/app/Interfaces/PostServiceInterface.php:16]: removeee!!

File: laravel-app/app/Services/PostService.php [hunk 3]
```diff
@@ -10,7 +10,6 @@ class PostService implements PostServiceInterface
 {
     /**
      * To create a new post.
-     * @param User $user
      * @param array<mixed, mixed> $validatedData
```
Comment (hipe01mark) [inline laravel-app/app/Services/PostService.php:13]: removee!!!!

File: laravel-app/app/Services/PostService.php [hunk 4]
```diff
@@ -25,10 +24,7 @@ public function createPost(User $user, array $validatedData): UserPost
 
     /**
      * To update an existing post.
-     *
-     * @param UserPost $post
      * @param array<string, mixed> $validatedData
-     * @return bool
      */
```
Comment (hipe01mark) [inline laravel-app/app/Services/PostService.php:28]: removee!!!! @

**Review by hipe01mark — COMMENTED** [review]
Check other methods if no comments and if there's a comment but there's @ on it. Remove it !!

File: laravel-app/app/Interfaces/PostServiceInterface.php (hunk 2 above)
Comment (adrean-hipe) [reply laravel-app/app/Interfaces/PostServiceInterface.php:16]: Apologies sir, but if I remove the @param for the arrray the larastan checker throws error and requires that to be added. 🙇

File: laravel-app/app/Services/PostService.php (hunk 4 above)
Comment (adrean-hipe) [reply laravel-app/app/Services/PostService.php:28]: Apologies sir, but if I remove the @param for the arrray the larastan checker throws error and requires that to be added. 🙇

File: laravel-app/app/Services/PostService.php (hunk 3 above)
Comment (adrean-hipe) [reply laravel-app/app/Services/PostService.php:13]: Already removed sir, as marked red by the git. Thanks 🚀

**Pull request closed without merging**